src/Auth/Actions/userActions.js

```diff
@@ -186,4 +186,61 @@ export const LogoutUser = (history) =>{
     
     history.push("/sign_in");
   };
-};
\ No newline at end of file
+};
+
+
+export const UploadProfilePIx = (formData, history) =>{
+    return async (dispatch) => {
+       const promise =   apiRequest('PATCH', `auth/v1/upload_pix`,formData,
+        {
+            headers: {
+                
+                "Content-Type": "multipart/form-data",
+                'Accept': 'multipart/form-data',
+            },
+        });
+         promise.then(
+            function (payload) {
+                const  {data}  = payload; 
+                if (data.status === "Failed") {
+                    toast.error('An Error occured!', {position: toast.POSITION.TOP_RIGHT});
+                    history.push('profile')
+                } else if (data.status === "success") {
+                    toast.success('Successful', {position: toast.POSITION.TOP_RIGHT});
+                }
+            },
+            function (error) {
+                throw new Error(error)
+            }
+        );
+        return promise;
+    }
+}
+
+export const UploadCoverPhoto = (formData, history) =>{
+    return async (dispatch) => {
+       const promise =   apiRequest('PATCH', `auth/v1/cover_photo`,formData,
+        {
+            headers: {
+                
+                "Content-Type": "multipart/form-data",
+                'Accept': 'multipart/form-data',
+            },
+        });
+         promise.then(
+            function (payload) {
+                const  {data}  = payload; 
```

Comment:
i don't understand this if else to check error. is the error not suppose to come in a catch block, and where is the catch block

Reply:
goodness please update all your actions.
`
export const UploadCoverPhoto = (formData, setCoverImg, setSubmitting) =>{
    return async (dispatch) => {
            const promise =   apiRequest('PATCH', `auth/v1/cover_photo`,formData,
                {
                    headers: {
                        "Content-Type": "multipart/form-data",
                        'Accept': 'multipart/form-data',
                    },
                });
                promise.then(payload => {
                        const  {data}  = payload; 
                            toast.success('Successful', {position: toast.POSITION.TOP_RIGHT});
                            dispatch(changePictureSuccess(data))
                    }).catch((error) => {
                      toast.error('An Error occured, Please try again', {position: toast.POSITION.TOP_RIGHT});
                     })
                  
                return promise;
    }
}
`